drivers: usb: udc: skeleton: share thread creation code across instances

mathieuchopstm opened the pull request:
Instead of having one "make_thread" function per instance, called through a function pointer in the instance configuration, save the thread stack information instead and consume it from shared code in the driver pre-init function. This makes the driver more readable (less code in the instance creation macro) and should reduce its footprint impact (since the thread creation code is no longer duplicated).

mathieuchopstm committed:
drivers: usb: udc: skeleton: share thread creation code across instances

Instead of having one "make_thread" function per instance, called through
a function pointer in the instance configuration, save the thread stack
information instead and consume it from shared code in the driver pre-init
function. This makes the driver more readable (less code in the instance
creation macro) and should reduce its footprint impact (since the thread
creation code is no longer duplicated).

Signed-off-by: Mathieu Choplain <mathieu.choplain-ext@st.com>

jfischer-no commented on drivers/usb/udc/udc_skeleton.c:
```diff
@@ -390,26 +404,6 @@ static const struct udc_api udc_skeleton_api = {
-		skeleton_thread_handler(dev);					\
-	}									\
-										\
-	static void udc_skeleton_make_thread_##n(const struct device *dev)	\
```
There was a reason for doing it this way, but I do not remember what it was. :crystal_ball:

mathieuchopstm replied:
Possibly to distinguish between threads for different UDCs? (but then I'd expect a `k_thread_set_name()`, and you can also do so using ELF symbols)

jfischer-no commented:
> `tests/drivers/udc` test `drivers.usb.udc.skeleton` does build the skeleton driver, but I don't know if CI builds it or not.
> 
> I did run the test locally to verify nothing was broken (although, in this instance, the thread is useless anyways...):

I thought it would be picked up for the PR run, but I cannot find it in the logs. Any ideas on how to fix that?

mathieuchopstm committed:
tests: drivers: udc: add integration platform for skeleton test

Add `integration_platforms` to the `drivers.usb.udc.skeleton` test case to
ensure the UDC Skeleton driver is built and tested by CI.

Signed-off-by: Mathieu Choplain <mathieu.choplain-ext@st.com>
